modules/pam_limits/pam_limits.c

```diff
@@ -1316,27 +1319,36 @@ pam_sm_open_session (pam_handle_t *pamh, int flags UNUSED,
             if (retval != PAM_SUCCESS)
                 break;
         }
+        last_filename = strdup(pl->conf_file);
+        if (!last_filename) {
+                pam_syslog(pamh, LOG_CRIT, "pam_sm_open_session: strdup failed");
+                return PAM_BUF_ERR;
+        }
+
         for (i = 0; filename_list[i] != NULL; i++)
             free(filename_list[i]);
         free(filename_list);
     }
 
     if (retval == PAM_IGNORE) {
-        D(("the configuration file ('%s') has an applicable '<domain> -' entry", pl->conf_file));
+        D(("the configuration file ('%s') has an applicable '<domain> -' entry", last_filename));
         free(pl->login_group);
+        free(last_filename);
         return PAM_SUCCESS;
     }
 
 out:
     if (retval != PAM_SUCCESS)
     {
-	pam_syslog(pamh, LOG_ERR, "error parsing the configuration file: '%s' ", pl->conf_file);
+	pam_syslog(pamh, LOG_ERR, "error parsing the configuration file: '%s' ", last_filename);
 	free(pl->login_group);
+	free(last_filename);
```

Comment:
At this point `last_filename` can still hold the value returned by `_pam_parse`, or the result of `pl->conf_file = LIMITS_FILE` assignment. In either of these cases it cannot be passed to `free`.